### ISSUE TITLE
Bandit terminating with missing :opts key

### ISSUE THREAD

**Issue by Hermanverschooten** [mtrudel/hap#57]
Hi,
I updated my dependencies in a Nerves project where I am using HAP successfully.
With the update it also upgraded `'ThousandIlands` and `Bandit`, now in my nerves device `Bandit` crashes with
```elixir
05:39:15.703 [error] GenServer #PID<0.2062.0> terminating
** (KeyError) key :opts not found in: %{plug: {HAP.HTTPServer, []}}
    (bandit 0.7.5) lib/bandit/http1/handler.ex:18: Bandit.HTTP1.Handler.handle_data/3
    (hap 0.4.7) /Users/herman/Projects/garage/deps/thousand_island/lib/thousand_island/handler.ex:343: HAP.HAPSessionHandler.handle_info/2
    (stdlib 4.3) gen_server.erl:1123: :gen_server.try_dispatch/4
    (stdlib 4.3) gen_server.erl:1200: :gen_server.handle_msg/6
    (stdlib 4.3) proc_lib.erl:240: :proc_lib.init_p_do_apply/3
Last message: {:tcp, #Port<0.307>, <<80, 79, 83, 84, 32, 47, 112, 97, 105, 114, 45, 118, 101, 114, 105, 102, 121, 32, 72, 84, 84, 80, 47, 49, 46, 49, 13, 10, 72, 111, 115, 116, 58, 32, 72, 101, 114, 109, 97, 110, 39, 115, 92, 48, 51, 50, 71, ...>>}
State: {%ThousandIsland.Socket{socket: #Port<0.307>, transport_module: HAP.HAPSessionTransport, read_timeout: 60000, span: %ThousandIsland.Telemetry{span_name: :connection, telemetry_span_context: #Reference<0.3405112061.268435457.195241>, start_time: 1779255243708, start_metadata: %{parent_telemetry_span_context: #Reference<0.3405112061.268435457.182629>, remote_address:
 {192, 168, 17, 65}, remote_port: 60529, telemetry_span_context: #Reference<0.3405112061.268435457.195241>}}}, %{plug: {HAP.HTTPServer, []}}}
```

I tried with the current master version, where I noticed you updated the child spec for Bandit, and that works.
Could you please release a new version?

Thanks,
Herman

**Comment by mtrudel** [comment]
Ah! I've been doing a bunch of work on Bandit/Thousand Island's startup & configuration options and yeah, this will be breaking

I'm trying to get those two to a 1.0 state in the next few weeks/months, which unfortunately means that there have been some breaking changes. I've got one more release of each that I'll be pushing out later today, and will come back to HAP as soon as they're out to get things working again. Will be by later today (EDT).

Sorry for the trouble!

**Comment by mtrudel** [comment]
Fixed in 0.4.8, just released!

**Comment by Hermanverschooten** [comment]
Great, thanks. I also proposed a PR on base36 to get rid of the warnings.
